src/controllers/eventController.ts

```diff
@@ -0,0 +1,195 @@
+import {Event} from '../entity/Event'
+import * as auth from '../middleware/authMiddleware';
+import {getManager} from 'typeorm';
+import {Request, Response} from 'express';
+import {validate} from 'class-validator';
+import { Resource } from '../entity/Resource';
+import {User} from '../entity/User';
+
+export const createEvent = async (req: Request, res: Response): Promise<void> => {
+    const eventRepository = getManager().getRepository(Event);
+    const resourceRepository = getManager().getRepository(Resource);
+
+    let eventData = req.body;
+    if(!eventData.startTime) {
+        res.status(400).send("Missing start time");
+    }
+    eventData.startTime = new Date(eventData.startTime);
+    if(eventData.endTime) {
+        eventData.endTime = new Date(eventData.endTime);
+    }
```

Comment:
Can these accept RFC3339 strings instead of unix timestamps?

Reply:
The Date constructor can take in a bunch of different formats, including ISO8601 (which I think is very similar to RFC3339, if not the same?). This is an example of a time I tested with: `2020-01-21T06:00:00.000Z`.

About the returning, if I understand correctly, when the Date object is sent by Express, it gets turned into JSON, which calls Date's toJSON(): https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toJSON
That calls toISOString(), which is again, ISO8601.

Reply:
Ok that sounds fine. We'll use ISO8601.